### PR TITLE
feat(dav): implement personal absence settings backend

### PR DESCRIPTION
**Pull request by st3iny**
## Summary

The backend for https://github.com/nextcloud/server/issues/40616.

All (related) feedback from https://github.com/nextcloud/server/pull/40767 is addressed here.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]
Looks good!

File: apps/dav/lib/Migration/Version1029Date20231004091403.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+			$table->addColumn('status', Types::STRING, [
+				'length' => 100,
+				'notnull' => true,
+			]);
+			$table->addColumn('message', Types::TEXT, [
+				'notnull' => true,
+			]);
+			$table->addUniqueIndex(['user_id'], 'dav_absence_uid_idx');
+		}
+		if ($table->getPrimaryKey() === null) {
```
[the remark this answers is not in the document]
Comment (ChristophWurst) [reply apps/dav/lib/Migration/Version1029Date20231004091403.php:74]: ^ is legit. $table is only defined if the first if body is run

[the remark this answers is not in the document]
Comment (st3iny) [reply apps/dav/lib/Migration/Version1029Date20231004091403.php:74]: Done

File: apps/dav/lib/Migration/Version1029Date20231004091403.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+			$table->setPrimaryKey(['id'], 'dav_absence_id_idx');
```
[the remark this answers is not in the document]
Comment (st3iny) [reply apps/dav/lib/Migration/Version1029Date20231004091403.php:75]: Done

**Review by JohannesGGE — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

File: apps/dav/appinfo/info.xml [hunk 3]
```diff
@@ -5,7 +5,7 @@
 	<name>WebDAV</name>
 	<summary>WebDAV endpoint</summary>
 	<description>WebDAV endpoint</description>
-	<version>1.28.0</version>
+	<version>1.29.0</version>
```
[the remark this answers is not in the document]
Comment (ChristophWurst) [reply apps/dav/appinfo/info.xml:8]: semantic > syntactic versioning :wink:

**Pull request merged**